accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -976,24 +976,18 @@ declaration itself, or we're talking about a particular generic
 instantiation of an extension type. *For non-generic extension type
 declarations, the representation type is the same in either case.*
 
-Let `V` be an extension type of the form
-<code>Name\<T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>, and let
-`R` be the corresponding instantiated representation type.  If `R` is
-non-nullable then `V` is a proper subtype of `Object`, and `V` is
-non-nullable.  Otherwise, `V` is a proper subtype of `Object?`, and
-`V` is potentially nullable.
+An extension type `V` is a proper subtype of `Object?`. It is potentially
+non-nullable, unless it implements `Object` or a subtype thereof
+*(as described in the section about extension types with superinterfaces)*.
 
 *That is, an expression of an extension type can be assigned to a top type
-(like all other expressions), and if the representation type is
-non-nullable then it can also be assigned to `Object`. Non-extension types
-(except bottom types and `dynamic`) cannot be assigned to extension types
-without an explicit cast. Similarly, null cannot be assigned to an
-extension type without an explicit cast, even in the case where the
-representation type is nullable (even better: don't use a cast, call a
-constructor instead). Another consequence of the fact that the extension
-type is potentially non-nullable is that it is an error to have an instance
-variable whose type is an extension type, and then relying on implicit
-initialization to null.*
+(like all other expressions). Non-extension types (except bottom types and
+`dynamic`) cannot be assigned to extension types without an explicit cast.
+Similarly, null cannot be assigned to an extension type without an explicit
```

Comment:
Complete nit: I'd code quote the "null":  `` `null` ``

Reply:
Actually, I'm using the word 'null' rather than the source code snippet '`null`' because the topic here is the semantic value associated with that syntax (and with many other code elements with varying syntactic forms, e.g., an expression whose type is of the form `T?`).

So the generality would go down, but the readability could of course go up.

Not sure. Haven't changed anything yet.

Reply:
In that case, use "the null value". The word "null" by itself is just too ambiguous by itself (IMO).

Reply:
Yes: We have used 'the null object' to make it obvious that it isn't syntax, I'll do that.